File: src/main/java/uk/gov/justice/hmpps/offendersearch/utils/JwtAuthInterceptor.java
package uk.gov.justice.hmpps.offendersearch.utils;

import org.springframework.http.HttpHeaders;
import org.springframework.http.HttpRequest;
import org.springframework.http.client.ClientHttpRequestExecution;
import org.springframework.http.client.ClientHttpRequestInterceptor;
import org.springframework.http.client.ClientHttpResponse;

import java.io.IOException;

public class JwtAuthInterceptor implements ClientHttpRequestInterceptor {

    @Override
    public ClientHttpResponse intercept(
            final HttpRequest request, final byte[] body, final ClientHttpRequestExecution execution)
            throws IOException {

        final var headers = request.getHeaders();
        headers.add(HttpHeaders.AUTHORIZATION, UserContext.getAuthToken());
        return execution.execute(request, body);
    }
}

File: src/main/kotlin/uk/gov/justice/hmpps/probationsearch/utils/JwtAuthInterceptor.kt
package uk.gov.justice.hmpps.probationsearch.utils

import org.springframework.http.HttpHeaders
import org.springframework.http.HttpRequest
import org.springframework.http.client.ClientHttpRequestExecution
import org.springframework.http.client.ClientHttpRequestInterceptor
import org.springframework.http.client.ClientHttpResponse

class JwtAuthInterceptor : ClientHttpRequestInterceptor {
  override fun intercept(
    request: HttpRequest,
    body: ByteArray,
    execution: ClientHttpRequestExecution
  ): ClientHttpResponse {
    val headers = request.headers
    headers.add(HttpHeaders.AUTHORIZATION, UserContext.getAuthToken())
    return execution.execute(request, body)
  }
}
